requirements.txt

```diff
@@ -0,0 +1,525 @@
+#
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
+#
+#    pip-compile --all-extras --resolver=backtracking
+#
+alabaster==0.7.12
+    # via sphinx
+anyio==3.6.2
+    # via jupyter-server
+argon2-cffi==21.3.0
+    # via jupyter-server
+argon2-cffi-bindings==21.2.0
+    # via argon2-cffi
+asteval==0.9.28
+    # via lmfit
+astor==0.8.1
+    # via flake8-simplify
+astropy==5.2
+    # via plasmapy (setup.py)
+asttokens==2.2.1
+    # via stack-data
+attrs==22.2.0
+    # via
+    #   flake8-implicit-str-concat
+    #   hypothesis
+    #   jsonschema
+    #   pytest
+babel==2.11.0
+    # via
+    #   jupyterlab-server
+    #   sphinx
+backcall==0.2.0
+    # via ipython
+beautifulsoup4==4.11.1
+    # via nbconvert
+bleach==5.0.1
+    # via nbconvert
+cached-property==1.5.2
+    # via plasmapy (setup.py)
+certifi==2022.12.7
+    # via requests
+cffi==1.15.1
+    # via argon2-cffi-bindings
+cfgv==3.3.1
+    # via pre-commit
+charset-normalizer==2.1.1
+    # via requests
+click==8.1.3
+    # via
+    #   click-default-group
+    #   towncrier
+    #   tryceratops
+click-default-group==1.2.2
+    # via towncrier
+codespell @ git+https://github.com/codespell-project/codespell/
+    # via plasmapy (setup.py)
+comm==0.1.2
+    # via ipykernel
+commonmark==0.9.1
+    # via rich
+contourpy==1.0.6
+    # via matplotlib
+cycler==0.11.0
+    # via matplotlib
+debugpy==1.6.4
+    # via ipykernel
+decorator==5.1.1
+    # via ipython
+defusedxml==0.7.1
+    # via nbconvert
+distlib==0.3.6
+    # via virtualenv
+dlint==0.13.0
+    # via plasmapy (setup.py)
+docutils==0.17.1
+    # via
+    #   nbsphinx
+    #   plasmapy (setup.py)
+    #   pybtex-docutils
+    #   restructuredtext-lint
+    #   sphinx
+    #   sphinx-rtd-theme
+    #   sphinxcontrib-bibtex
```

Comment:
It looks like there are some `docs` dependencies mixed in here.  Perhaps we could remove those for the Python 3.10 tests.

Reply:
Yeah, I'm not sure how best to do this:
- Using two lockfiles means they might, theoretically, get out of sync (if a doc dependency has an upper bound on a test dependency, for example).
- But using a single one with `--all-extras` means the install get larger (which could be sidestepped to a degree by finally setting up proper caching for test envs in github actions).
- Using a single lockfile just for the test extra means we aren't catching any doc issues. That's still helpful, then, just less so.